### ISSUE TITLE
How to train the model on custom data?

### ISSUE THREAD

**Issue by diwashrestha** [wvuvl/SiamABC#6]
I'm trying to train the model on my own custom dataset. I followed the training instructions provided in this repository and in FearTracker. I converted my dataset into CSV format as instructed and was able to successfully run the training pipeline.

However, I am facing two main issues:

**Poor Performance:**
The output from the trained model is not satisfactory. I suspect there might be an issue with how I’ve prepared or formatted my training data, but I'm not sure what exactly could be going wrong. Could you clarify any important aspects of data preparation that are easy to overlook?

**Large Model Size:**
I noticed that the size of the fine-tuned model I obtained is significantly larger than the official S_Small and S_Tiny checkpoints. Is this expected behaviour when fine-tuning on a custom dataset, or might I have accidentally changed some configuration that affects the model size?

**Comment by rz0012** [comment]
Thank you for your interest. And I apologize for the late response. 

To address your first concern, there are many components that can affect the training:
- If the custom data is small, adjusting the learning rate or having a pre-trained tracking model might be useful.
- On the dataloading step, were you able to visualize if the preprocessed data is correctly producing bounding boxes and the object appears in the frame?
- If the previous step is correct, then you could potentially try visualizing directly from the dataloader to see if that is correctly being loaded. 
- In `https://github.com/wvuvl/SiamABC/blob/master/core/utils/sampler.py`, `self.frame_offset = 150`, you could potentially change this value to reflect your custom dataset. 
- There might be other optimization-related issues that could have been overlooked. 

Please let me know if they make sense. 

To address your second question:
- During training we use projection heads similar to SimSiam, and other self-supervised training algorithms. These projection heads are heavy in terms of parameter space. However, they serve no purpose once the model is trained; therefore, we only use the parameters that are being used for the tracking and eliminate those that are not. Therefore, the end model size is significantly less. 
- Please let me know if that makes sense.

**Comment by diwashrestha** [comment]
Thank you for your Response. I would like to apologise for the late reply.

I will go through the steps you have suggested for the fine-tuning and will follow up with the results.

Regarding the model optimisation, is there a separate post-processing script that removes the parameters unrelated to tracking from the final fine-tuned model? Or does the fine-tuning pipeline handle this automatically and output a streamlined version that retains only the tracking-specific parameters?

**Comment by diwashrestha** [comment]
How to use the pretrained model during training? 

I think the changes should be made in the SiamABC.yaml  in core>config>model.

```_target_: core.models.SiamABC.SiamABCNet
backbone: custom_fbnet
growth_factor: 1.2
upsample: pixel_shuffle
pretrained: true
num_classes: 1
num_filters: 32
num_channels: 3
align: false
img_size: 256
stride: 2
conv_block: sep_conv
towernum: 2
model_size: M
max_layer: 4
crop_template_features: False
pretrained_path: assets/S_Small/model_S_Small_v1.pt 
```

**Comment by rz0012** [comment]
I apologize for the late reply. Unfortunately, our current script does not explicitly handle loading a pretrained model for training. However, it could easily be integrated into the https://github.com/wvuvl/SiamABC/blob/master/train.py file after the model is initialized.

Regarding removing unnecessary parameters during tracking, I apologize that we did not make that script available. However, if you are still experiencing these issues, please let me know, and I can forward the relevant scripts to you. Thank you for your patience.

**Comment by BRomanchuk** [comment]
Hi!

I faced the same problem with the model size. Can you please share the post-processing script you mentioned?
